src/exposurelog/routers/add_message.py

```diff
@@ -4,19 +4,22 @@
 import collections.abc
 import http
 import logging
+import re
 
-import astropy.time
-import astropy.units as u
 import fastapi
+import lsst.daf.butler
 import lsst.daf.butler.registry
 import sqlalchemy as sa
 
 from ..message import ExposureFlag, Message
 from ..shared_state import SharedState, get_shared_state
+from ..utils import current_date_and_day_obs
 from .normalize_tags import TAG_DESCRIPTION, normalize_tags
 
 router = fastapi.APIRouter()
 
+OBSID_REGEX = re.compile(r"[A-Z][A-Z]_[A-Z]_(\d\d\d\d\d\d\d\d)_(\d\d\d\d\d\d)")
```

Comment:
Do you want a `$` at the end of the regex (and `^` at start) to ensure that you are matching a complete obsid? Consider `r"^[A-Z][A-Z]_[A-Z]_(\d{8})_(\d{6})$"` rather than repeating the `\d`.

Reply:
I'm using `fullmatch` so there is no need.

Reply:
Please add an inline comment to indicate that the intended use is with `fullmatch` so that's clear to anyone reading the code.